docs/topics/iotea-components.md

```diff
@@ -8,68 +8,132 @@
 
-- Freely scalable by ingestion instance count
-- Uses rules to determine whether an event should be forwareded to a specific talent
-  - Rules can be any type of boolean algebra (containing and/or rules)
-  - Enriches event with all fields a talent is "interested" in
+## Ingestion
 
-### Talent
+This component is used to transform and validate incoming events and it's configurable as an ETL-pipeline and Multi-Channel supports an arbitrary amount of parallel ETL-pipelines and its completely configurable via configuration files (json) and scalable via multiple instances.
 
-- Supports local and cloud talents (using FAAS like Lambda functions)
-  - Automatically syncs to cloud message brokers via ProtocolGateway
+It validates value types (boolean, string, number, object, any) and _Types_ against metadata (e.g. using [_MetadataManager_](#MetadataManager)).
+
+## Encoding
+
+This component encodes incoming events by various encoders (minmax, delta, categorical) and it uses the InstanceManager to store an encoded field into the appropriate instance. It's scalable via multiple instances.
+
+## Routing
+
+This component routes the events to specific [_Talents_](#Talent) by evaluating their related rule-set. It's scalable via multiple instances.
+
+## Sub Components
+
+This components are used by the other platform components and act like helper or has been extracted to make them re-usable.
+
+### MetadataManager
+
+This component holds the definition of segments, feature and types and will be synchronized asynchronously using the underlying publish-subscribe protocol.
+
+It synchronizes its internal state via publish-subscribe with all other InstanceManager instances.
+
+### TalentConfigManager
+
+This component holds the configuration of [_Talents_](#Talent) like the rules which are used to subscribe to _Feature_ events.
+
+It synchronizes its internal state via publish-subscribe with all other InstanceManager instances.
+
+### InstanceManager
+
+This component keeps track of all instances of a given subject (An instance is a specific device of a specific type belonging to a specific segment). It provides a TTL for every field to simulate a rolling time window.
+
+It synchronizes its internal state via publish-subscribe with all other InstanceManager instances.
+
+### Rules
+
+This component provides AND & OR rules using one of the following constraints:
+
+- isSet, change, greater than, less than, ...
+- time series constraints
+- Nelson constraints
+
+### ProtocolGateway
+
+The ProtocolGateway abstracts the publish-subscribe protocol from a specific implementation of it. Therefore, a _ProtocolGatewayAdapter_ is used to integrate the underlying publish-subscribe communication (e.g. MQTT Broker).
+
+This abstraction is not limited for platform components. It is also the common way to abstract the clients ([_Talents_](#Talent)).
```

Comment:
to abstract the clients (Talents)
  suggestion -->
to abstract the clients (Talents) communication

Reply:
Done